### ISSUE TITLE
[SC2031] false positive (from `for PATH_W in ${PATH}; do`)

### ISSUE THREAD

**Issue by SwuduSusuwu** [koalaman/shellcheck#3112]
#### For bugs
- Rule Id: SC2031
- My shellcheck version: https://github.com/SwuduSusuwu/SubStack/blob/trunk/.github/workflows/codacy.yml
- [ X ] The rule's wiki page does not already cover this (e.g. https://shellcheck.net/wiki/SC2031)
- [ X ] I tried on https://www.shellcheck.net/ and verified that this is still a problem on the latest commit

#### Here's a snippet or screenshot that shows the problem:
https://github.com/SwuduSusuwu/SubStack/blob/27721b2f195b598ac25fd052af0d27ca5ac0c48b/Macros.sh#L351-L402

#### Here's what shellcheck currently says:
> [PATH was modified in a subshell. That change might be lost.](https://github.com/SwuduSusuwu/SubStack/security/code-scanning/2143)

Before this, it was [In _POSIX_ sh, `local` is undefined.](https://www.shellcheck.net/wiki/SC3043)

#### Here's what I wanted or expected to see:

> All checks passed

**Comment by SwuduSusuwu** [comment]
It appears that this triggers whenever you follow the suggestion to replace `local` with subshells, and that the sole solution is to replace `#!/bin/sh` with
```
#!/bin/sh
#shellcheck disable=SC2031 #This triggers whenever you follow the suggestion to replace `local` with subshells (`$()`). 
```

**Comment by SwuduSusuwu** [comment]
[SC2031 even triggers for `if [ -z "${PATH}" ]; then`](https://github.com/SwuduSusuwu/SubStack/security/code-scanning/2144)

**Comment by SwuduSusuwu** [comment]
This appears to have resolved on its own.
